### ISSUE TITLE
Cannot create FunctionFailedException in unit test

### ISSUE THREAD

**Issue by DarranShepherd** [Azure/azure-functions-durable-extension#753]
I am creating unit tests for my orchestrator function, following the guidance on https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-unit-testing .

I wish to test the logic that handles one of my activity tasks throwing an exception. The orchestrator  handles this with
```C#
catch (FunctionFailedException ex) when (ex.InnerException is MyCustomException)
```

However, I am unable to mock the call to `CallActivityAsync` and have it throw as all constructors of `FunctionFailedException` are internal. As `Exception.InnerException` is not virtual, I also cannot use `Mock.Of<FunctionFailedException>(ex => ex.InnerException == new MyCustomException())`

I have been able to resolve this temporarily by using reflection, but I doubt this is the expected approach. Is this not something that should be tested for some reason, or is there an official way to do so?

```C#
private FunctionFailedException CreateException(string message, Exception inner)
{
    var ctor = typeof(FunctionFailedException).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, new[] { typeof(string), typeof(Exception) }, null);
    return (FunctionFailedException)ctor.Invoke(new object[] { message, inner });
}
```

**Comment by cgillum** [comment]
I think it makes sense for us to expose those constructors to make things more unit testable. @ConnorMcMahon what do you think?

**Comment by cgillum** [comment]
This fix was released as part of [v1.8.2](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.2).
